tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp

```diff
@@ -0,0 +1,4508 @@
+  } else {
+    STORAGE_LOG(DEBUG, "succeed to process ObAdminTabletMetaValidationTask", K(ret), K(task_id_));
+  }
+  return ret;
+}
+int ObAdminTabletMetaValidationTask::collect_tablet_meta_info_()
+{
+  int ret = OB_SUCCESS;
+  void *alc_ptr = nullptr;
+  ObAdminBackupTabletValidationDag *tablet_validation_dag = nullptr;
+  ObAdminBackupSetValidationAttr *backup_set_attr = nullptr;
+  ObArray<ObAdminBackupTabletValidationAttr *> current_group;
+  if (IS_NOT_INIT) {
+    ret = OB_NOT_INIT;
+    STORAGE_LOG(WARN, "not init", K(ret));
+  } else if (OB_ISNULL(tablet_validation_dag
+                       = static_cast<ObAdminBackupTabletValidationDag *>(get_dag()))) {
+    ret = OB_ERR_UNEXPECTED;
+    STORAGE_LOG(WARN, "failed to get tablet validation dag", K(ret));
+  } else if (OB_FAIL(tablet_validation_dag->processing_tablet_group_.at(task_id_, current_group))) {
+    ret = OB_ERR_UNEXPECTED;
+    STORAGE_LOG(WARN, "failed to get tablet tablet group", K(ret));
+  } else if (OB_FAIL(tablet_meta_.reserve(current_group.count()))) {
+    STORAGE_LOG(WARN, "failed to reserve tablet meta", K(ret));
+  } else {
+    for (int64_t i = 0; OB_SUCC(ret) && i < current_group.count(); i++) {
+      const ObAdminBackupTabletValidationAttr *tablet_attr = nullptr;
+      int64_t backup_set_id = -1;
+      if (OB_ISNULL(tablet_attr = current_group.at(i))) {
+        ret = OB_ERR_UNEXPECTED;
+        STORAGE_LOG(WARN, "tablet attr is null", K(ret));
+      } else if (FALSE_IT(backup_set_id = tablet_attr->tablet_meta_index_->backup_set_id_)) {
+      } else if (OB_FAIL(ctx_->get_backup_set_attr(backup_set_id, backup_set_attr))) {
+        STORAGE_LOG(WARN, "failed to get backup set attr", K(ret), K(backup_set_id));
+      } else if (OB_ISNULL(backup_set_attr)) {
+        ret = OB_ERR_UNEXPECTED;
+        STORAGE_LOG(WARN, "backup set attr is null", K(ret), K(backup_set_id));
+      } else if (OB_ISNULL(alc_ptr = local_allocator_.alloc(sizeof(backup::ObBackupTabletMeta)))) {
+        ret = OB_ALLOCATE_MEMORY_FAILED;
+        STORAGE_LOG(WARN, "failed to alloc backup::ObBackupTabletMeta", K(ret));
+      } else if (FALSE_IT(tablet_meta_.at(i) = new (alc_ptr) backup::ObBackupTabletMeta())) {
+      } else if (OB_FAIL(ObAdminBackupValidationUtil::read_tablet_meta(
+                     backup_set_attr->backup_set_store_->get_backup_set_dest(),
+                     *tablet_attr->tablet_meta_index_, tablet_attr->data_type_, *tablet_meta_.at(i),
+                     ctx_))) {
+        STORAGE_LOG(WARN, "failed to read tablet meta", K(ret), K(backup_set_id),
+                    K(tablet_attr->tablet_meta_index_), K(tablet_attr->data_type_));
+      } else if (!tablet_meta_.at(i)->tablet_meta_.ha_status_.is_none()) {
+        ret = OB_ERR_UNEXPECTED;
+        STORAGE_LOG(WARN, "tablet meta is not complete", K(ret), K(backup_set_id),
+                    K(*tablet_meta_.at(i)));
+      }
+    }
+    if (OB_SUCC(ret)) {
+      if (current_group.count() == 2) {
```

Comment:
const put before hand

Reply:
the other side of == is not assignable. should do that?

Reply:
all fixed